### PR TITLE
Improve rand(bool)

### PR DESCRIPTION
**Pull request by planetis-m**
Improves the code generated for rand(bool).

### CODE REVIEW

**Comment by planetis-m** [comment]
tconvexhull.nim test failure seems unrelated.

File: lib/pure/random.nim [hunk 1]
```diff
@@ -377,8 +377,10 @@ proc rand*[T: Ordinal](t: typedesc[T]): T =
       assert rand(range[1..16]) in 1..16
   # pending csources >= 1.4.0 or fixing https://github.com/timotheecour/Nim/issues/251#issuecomment-831599772,
   # use `runnableExamples("-r:off")` instead of `if false`
-  when T is range or T is enum or T is bool:
+  when T is range or T is enum:
     result = rand(state, low(T)..high(T))
+  elif T is bool:
+    result = (1 and state.next) == 1
```
Comment (rockcavera) [inline lib/pure/random.nim]: Testing whether the integer is even or odd is not a good idea for booleans when it comes to this prng algorithm. The author himself recommends doing the sign test to obtain a random boolean. See https://xoroshiro.di.unimi.it/xoroshiro128plus.c

Comment (planetis-m) [reply lib/pure/random.nim]: > and right shifts to extract subsets of bits.

Should we change the casts as well?

**Comment by planetis-m** [comment]
With the sign test and compiling to JS, rand always returns false.

```js
function rand_469762050() {
  var result_469762054 = false;
  result_469762054 = (next_603979819(state_603979789[0]) < 0);
  return result_469762054;
}
```

**Comment by planetis-m** [comment]
Does that need to branches (js, not js), or should I revert and call it a day?

**Comment by planetis-m** [comment]
Ready.

**Comment by planetis-m** [comment]
Sorry I was looking at the wrong function disassembly.

File: lib/pure/random.nim [hunk 2]
```diff
@@ -377,10 +377,12 @@ proc rand*[T: Ordinal](t: typedesc[T]): T =
...
   # pending csources >= 1.4.0 or fixing https://github.com/timotheecour/Nim/issues/251#issuecomment-831599772,
   # use `runnableExamples("-r:off")` instead of `if false`
-  when T is range or T is enum or T is bool:
+  when T is range or T is enum:
     result = rand(state, low(T)..high(T))
+  elif T is bool:
+    result = bool(state.next shr 63)
   else:
-    result = cast[T](state.next)
+    result = cast[T](state.next shr (64 - sizeof(T)*8))
```
Comment (planetis-m) [inline lib/pure/random.nim:385]: Right shifting by zero doesn't generate any code, so it seems ok.

**Comment by planetis-m** [comment]
@Varriount it's ready.

**Review by rockcavera — APPROVED** [review]

**Pull request merged**